instrumentation/hibernate/hibernate-6.0/src/main/java/com/solarwinds/opentelemetry/instrumentation/hibernate/v6_0/DrsaInstrumentation.java

```diff
@@ -0,0 +1,127 @@
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.hibernate.sql.results.jdbc.internal.DeferredResultSetAccess;
+
+public class DrsaInstrumentation implements TypeInstrumentation {
+
+  @Override
+  public ElementMatcher<ClassLoader> classLoaderOptimization() {
+    return hasClassesNamed("org.hibernate.sql.results.jdbc.internal.DeferredResultSetAccess");
+  }
+
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return named("org.hibernate.sql.results.jdbc.internal.DeferredResultSetAccess");
+  }
+
+  @Override
+  public void transform(TypeTransformer transformer) {
+    transformer.applyAdviceToMethod(
+        isMethod().and(named("executeQuery")),
+        DrsaInstrumentation.class.getName() + "$DeferredResultSetAccessAdvice");
+  }
+
+  @SuppressWarnings("unused")
+  public static class DeferredResultSetAccessAdvice {
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void startMethod(
+        @Advice.This DeferredResultSetAccess drsa,
+        @Advice.Local("swoCallDepth") CallDepth callDepth,
+        @Advice.Local("swoSqlContext") String swoSql,
+        @Advice.Local("swoContext") Context context,
+        @Advice.Local("swoScope") Scope scope) {
+      callDepth = CallDepth.forClass(DeferredResultSetAccess.class);
+      if (callDepth.getAndIncrement() > 0) {
+        return;
+      }
+
+      try {
+        // We're using reflection here because this seems to be the best place we can intercept the
+        // raw sql before the PreparedStatement is generated and be able to make the span a parent
+        // of the execution step.
+
+        Class<?> clazz = drsa.getClass();
+        Field finalSql = clazz.getDeclaredField("finalSql");
+        finalSql.setAccessible(true);
+
+        String sql = (String) finalSql.get(drsa);
+        finalSql.setAccessible(false);
+        Context parentContext = currentContext();
+        if (!HibernateInstrumenter.getInstance().shouldStart(parentContext, sql)) {
+          return;
+        }
+
+        Field statementCreator = clazz.getDeclaredField("statementCreator");
+        statementCreator.setAccessible(true);
+        Function<String, PreparedStatement> function =
+            (Function<String, PreparedStatement>) statementCreator.get(drsa);
+
+        if (!(function instanceof FunctionWrapper)) {
+          // We wrap the `statementCreator` field with our own wrapper to inject the context. This
+          // is less risky compared to modifying `finalSql` field.
+          statementCreator.set(drsa, new FunctionWrapper(function));
+        }
+        statementCreator.setAccessible(false);
+
+        context = HibernateInstrumenter.getInstance().start(parentContext, sql);
+        scope = context.makeCurrent();
+        swoSql = sql;
+
+      } catch (Throwable ignore) {
+        // ignore because we can't do jack here
```

Comment:
could log a message if it's safe and if the info could help diagnose instrumentation issue ;)